Client/src/Components/Admin/AdminReviewView.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box, Button, Grid, TextField, Typography, createMuiTheme, withStyles } from '@material-ui/core'
+import { IApprovedDatasetModel, IFlaggedDatasetQuery } from '../../Models/Datasets/IApprovedDatasetModel'
+import React, { useEffect, useState } from 'react'
+import { adminApprovedDataset, callRejectDataset, flagDataset } from '../../Remote/Endpoints/DatasetEndpoints'
+
+import { AdminReviewList } from './AdminReviewList'
+import { DatasetUploadForm } from '../DatasetUpload/DatasetUploadForm'
+
+export function AdminReviewView() {
+
+    const [datasetState, setDatasetState] = useState([])
+    const [editable, setEditable] = useState(false)
+    const [dataset, setDataset] = useState<IApprovedDatasetModel>()
+    const [comment, setComment] = useState("")
+    const [flaggedComment, setFlaggedComment] = useState("")
+    const [update, setUpdate] = useState(0)
+
+    const handleDeleteDataset = async () => {
+        await callRejectDataset(dataset.id)
+        setUpdate(update + 1)
+        handleDatasetChange(null)
+    }
+
+    const handleEditDataset = () => {
+        setEditable(!editable)
+    }
+
+    const handleFlagDataset = async () => {
+        const query: IFlaggedDatasetQuery = { datasetId: dataset.id, flaggedComments: flaggedComment, additionalComments: comment }
+        await flagDataset(query)
+    }
+
+    const handleApproveDataset = async (datasetId: IApprovedDatasetModel) => {
+        const query: IFlaggedDatasetQuery = { datasetId: dataset.id, additionalComments: comment }
+        await adminApprovedDataset(query)
+        setUpdate(update + 1)
```

Comment:
We should make sure to add snackbar messages confirming approval, flagging and removal

Reply:
Snackbar has been implemented now.

Reply:
Awesome